homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -73,6 +78,17 @@ async def async_step_user(
                 errors["base"] = "cannot_connect"
 
             if info:
+                if self._reauth_entry:
+                    self.hass.config_entries.async_update_entry(
+                        self._reauth_entry, data=user_input
```

Comment:
What happens to CONF_REFRESH_TOKEN? I don't see that specified in `user_input` here, but it is when the entry is initially created so i'm worried it gets removed.

e.g. in the tests FIXTURE_CONFIG_ENTRY contains other data fields that are not reflected in FIXTURE_USER_INPUT so that seems like a bug.

Reply:
Good catch, thanks! I've adjusted the code so `CONF_REFRESH_TOKEN` is always set. 

Didn't catch it live as without a refresh token (or an invalid one), the library itself will failover and login using username and password. However the original reason to implement it was to reduce the amount of user/password logins and increase the number of refresh_token logins, so it definetly makes sense.